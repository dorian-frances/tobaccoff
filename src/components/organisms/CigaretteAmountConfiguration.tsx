import { StyleProp, StyleSheet, View, ViewStyle } from 'react-native';
import React from 'react';
import SectionText from '../atoms/texts/SectionText';
import SegmentedButtonsTobaccoType from '../molecules/SegmentedButtonsTobaccoType';
import InputText from '../atoms/texts/InputText';

type ConfigurationCigaretteAmountProps = {
  containerStyle?: StyleProp<ViewStyle>;
};

const CigaretteAmountConfiguration = ({
  containerStyle = styles.container,
}: ConfigurationCigaretteAmountProps) => {
  return (
    <View style={containerStyle}>
      <View style={styles.sectionTextStyle}>
        <SectionText text={'Je fumais : '} />
      </View>
      <View style={styles.segmentedButtonStyle}>
        <SegmentedButtonsTobaccoType />
      </View>
      <View style={styles.inputTextStyle}>
        <InputText label={'Cigarettes / jours'} rightText={'/jours'} />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    width: '100%',
  },
  sectionTextStyle: {
    alignItems: 'center',
    marginBottom: 20,
  },
  segmentedButtonStyle: {
    marginBottom: 20,
  },
  inputTextStyle: {},
});

export default CigaretteAmountConfiguration;
