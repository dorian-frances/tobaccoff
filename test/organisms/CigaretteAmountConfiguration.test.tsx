import React from 'react';
import { render } from '@testing-library/react-native';
import CigaretteAmountConfiguration from '../../src/components/organisms/CigaretteAmountConfiguration';

describe('<CigaretteAmountConfiguration/>', () => {
  it('should render correctly', () => {
    const fakeCigaretteAmountRetriever = jest.fn();
    const fakeCigaretteTypeRetriever = jest.fn();
    const component = render(
      <CigaretteAmountConfiguration
        getCigaretteAmount={fakeCigaretteAmountRetriever}
        getCigaretteType={fakeCigaretteTypeRetriever}
      />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
