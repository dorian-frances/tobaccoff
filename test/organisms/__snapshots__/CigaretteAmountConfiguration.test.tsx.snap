// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CigaretteAmountConfiguration/> should render correctly 1`] = `
<View>
  <View
    style={
      {
        "alignItems": "center",
        "marginBottom": 20,
      }
    }
  >
    <Text
      style={
        {
          "color": "#000000",
          "fontFamily": "Montserrat-Medium",
          "fontSize": 22,
        }
      }
    >
      Je fumais : 
    </Text>
  </View>
  <View
    style={
      {
        "marginBottom": 20,
      }
    }
  >
    <View
      style={
        [
          {
            "flexDirection": "row",
          },
          undefined,
        ]
      }
    >
      <View
        style={
          [
            {
              "backgroundColor": "rgba(232, 222, 248, 1)",
              "borderBottomRightRadius": 0,
              "borderColor": "rgba(121, 116, 126, 1)",
              "borderRadius": 20,
              "borderRightWidth": 0,
              "borderTopRightRadius": 0,
              "borderWidth": 1,
            },
            {
              "borderStyle": "solid",
              "flex": 1,
              "minWidth": 76,
            },
            [
              undefined,
              {},
            ],
          ]
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": true,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              false,
              {
                "borderBottomRightRadius": 0,
                "borderRadius": 20,
                "borderRightWidth": 0,
                "borderTopRightRadius": 0,
              },
            ]
          }
          testID="industrialTypeButton"
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "paddingHorizontal": 16,
                  "paddingVertical": 9,
                },
                {
                  "paddingVertical": 9,
                },
              ]
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "marginRight": 5,
                  "transform": [
                    {
                      "scale": 0,
                    },
                  ],
                }
              }
              testID="industrialTypeButton-check-icon"
            >
              <Text />
            </View>
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                [
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "letterSpacing": 0.1,
                    "lineHeight": 20,
                  },
                  {
                    "textAlign": "left",
                  },
                  {
                    "writingDirection": "ltr",
                  },
                  [
                    {
                      "textAlign": "center",
                    },
                    {
                      "color": "rgba(28, 27, 31, 1)",
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                  ],
                ]
              }
            >
              Industrielles
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderBottomLeftRadius": 0,
              "borderColor": "rgba(121, 116, 126, 1)",
              "borderRadius": 20,
              "borderTopLeftRadius": 0,
              "borderWidth": 1,
            },
            {
              "borderStyle": "solid",
              "flex": 1,
              "minWidth": 76,
            },
            [
              undefined,
              {},
            ],
          ]
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": false,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              false,
              {
                "borderBottomLeftRadius": 0,
                "borderRadius": 20,
                "borderTopLeftRadius": 0,
              },
            ]
          }
          testID="rolledTypeButton"
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "paddingHorizontal": 16,
                  "paddingVertical": 9,
                },
                {
                  "paddingVertical": 9,
                },
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                [
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "letterSpacing": 0.1,
                    "lineHeight": 20,
                  },
                  {
                    "textAlign": "left",
                  },
                  {
                    "writingDirection": "ltr",
                  },
                  [
                    {
                      "textAlign": "center",
                    },
                    {
                      "color": "rgba(28, 27, 31, 1)",
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                  ],
                ]
              }
            >
              Roul√©es
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={{}}
  >
    <View
      style={
        {
          "fontFamily": "Montserrat-Medium",
        }
      }
    >
      <View
        pointerEvents="none"
        style={
          [
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 6,
            },
            {
              "backgroundColor": "rgba(255, 251, 254, 1)",
              "borderColor": "rgba(121, 116, 126, 1)",
              "borderRadius": 4,
              "borderWidth": 1,
            },
            {
              "borderColor": "#79747E",
              "borderRadius": 100,
              "width": "100%",
            },
          ]
        }
        testID="text-input-outline"
      />
      <View>
        <View
          style={
            [
              {
                "paddingBottom": 0,
              },
              {
                "minHeight": 56,
                "paddingTop": 8,
              },
            ]
          }
        >
          <View
            collapsable={false}
            pointerEvents="none"
            style={
              {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "transform": [
                  {
                    "translateX": 0,
                  },
                ],
                "zIndex": 3,
              }
            }
          >
            <Text
              collapsable={false}
              maxFontSizeMultiplier={1.5}
              numberOfLines={1}
              style={
                {
                  "backgroundColor": "rgba(255, 251, 254, 1)",
                  "color": "transparent",
                  "fontFamily": "System",
                  "fontSize": 15,
                  "fontWeight": undefined,
                  "left": 8,
                  "letterSpacing": 0.15,
                  "lineHeight": undefined,
                  "opacity": -0.6666666666666667,
                  "paddingHorizontal": 0,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 33,
                  "transform": [
                    {
                      "translateX": 0,
                    },
                    {
                      "translateY": 0,
                    },
                    {
                      "scale": 1,
                    },
                    {
                      "scaleY": 1,
                    },
                  ],
                  "width": -16,
                  "writingDirection": "ltr",
                }
              }
            >
              Cigarettes / jour
            </Text>
            <Text
              collapsable={false}
              maxFontSizeMultiplier={1.5}
              numberOfLines={1}
              onLayout={[Function]}
              style={
                {
                  "color": "rgba(103, 80, 164, 1)",
                  "fontFamily": "System",
                  "fontSize": 15,
                  "fontWeight": undefined,
                  "left": 0,
                  "letterSpacing": 0.15,
                  "lineHeight": undefined,
                  "opacity": 0,
                  "paddingHorizontal": 16,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 32,
                  "transform": [
                    {
                      "translateX": 0,
                    },
                    {
                      "translateY": 0,
                    },
                    {
                      "scale": 1,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
              testID="text-input-ID-label-active"
            >
              Cigarettes / jour
            </Text>
            <Text
              collapsable={false}
              maxFontSizeMultiplier={1.5}
              numberOfLines={1}
              style={
                {
                  "color": "rgba(73, 69, 79, 1)",
                  "fontFamily": "System",
                  "fontSize": 15,
                  "fontWeight": undefined,
                  "left": 0,
                  "letterSpacing": 0.15,
                  "lineHeight": undefined,
                  "opacity": 0,
                  "paddingHorizontal": 16,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 32,
                  "transform": [
                    {
                      "translateX": 0,
                    },
                    {
                      "translateY": 0,
                    },
                    {
                      "scale": 1,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
              testID="text-input-ID-label-inactive"
            >
              Cigarettes / jour
            </Text>
          </View>
          <TextInput
            editable={true}
            keyboardType="numeric"
            maxFontSizeMultiplier={1.5}
            multiline={false}
            onBlur={[Function]}
            onChangeText={[Function]}
            onEndEditing={[Function]}
            onFocus={[Function]}
            placeholder=" "
            placeholderTextColor="rgba(73, 69, 79, 1)"
            selectionColor="rgba(103, 80, 164, 1)"
            style={
              [
                {
                  "margin": 0,
                  "zIndex": 1,
                },
                {
                  "height": 48,
                },
                {
                  "paddingBottom": 0,
                  "paddingTop": 0,
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "fontFamily": "System",
                  "fontSize": 15,
                  "fontWeight": undefined,
                  "letterSpacing": 0.15,
                  "lineHeight": undefined,
                  "paddingHorizontal": 16,
                  "textAlign": "left",
                  "textAlignVertical": "center",
                },
                false,
                {
                  "marginRight": 0,
                  "paddingRight": 56,
                },
                undefined,
              ]
            }
            testID="text-input-ID"
            underlineColorAndroid="transparent"
          />
        </View>
        <View
          collapsable={false}
          onLayout={[Function]}
          style={
            {
              "alignItems": "center",
              "justifyContent": "center",
              "opacity": 0,
              "position": "absolute",
              "right": 16,
              "top": 31,
            }
          }
          testID="right-affix-adornment"
        >
          <Text
            maxFontSizeMultiplier={1.5}
            style={
              [
                {
                  "color": "rgba(28, 27, 31, 0.54)",
                },
                {
                  "fontFamily": "System",
                  "fontSize": 15,
                  "fontWeight": undefined,
                  "letterSpacing": 0.15,
                  "lineHeight": undefined,
                },
                undefined,
              ]
            }
            testID="right-affix-adornment-text"
          >
            /jour
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
